server/src/main/java/org/elasticsearch/action/ActionListenerResponseHandler.java

```diff
@@ -24,7 +24,7 @@
  */
 public class ActionListenerResponseHandler<Response extends TransportResponse> implements TransportResponseHandler<Response> {
 
-    private final ActionListener<? super Response> listener;
+    protected final ActionListener<? super Response> listener;
```

Comment:
Is this change required? It looks like all of the updated usages are referencing same listener passed in constructor or am I missing something?

Reply:
This way we save a reference in some, because we don't have to capture `listener` twice (once in the variable and once in the anonymous capture). We have the same approach in `ActionRunnable` so I did it here also to save some references (if only to make heap dumps easier to grind through :D).